src/Fees.sol

```diff
@@ -70,18 +75,34 @@ abstract contract Fees is IFees, Owned {
         if (dynamicSwapFee >= MAX_SWAP_FEE) revert FeeTooLarge();
     }
 
-    /// @dev Only the lower 12 bits are used here to encode the fee denominator.
-    function _checkProtocolFee(uint16 fee) internal pure {
-        if (fee != 0) {
-            uint16 fee0 = fee % 64;
-            uint16 fee1 = fee >> 6;
-            // The fee is specified as a denominator so it cannot be LESS than the MIN_PROTOCOL_FEE_DENOMINATOR (unless it is 0).
-            if (
-                (fee0 != 0 && fee0 < MIN_PROTOCOL_FEE_DENOMINATOR) || (fee1 != 0 && fee1 < MIN_PROTOCOL_FEE_DENOMINATOR)
-            ) {
-                revert FeeTooLarge();
+    function _isValidProtocolFees(uint24 protocolFees) internal pure returns (bool) {
+        if (protocolFees != 0) {
```

Comment:
this is all repeated code. Fine to expose _isValidProtocolFees but then we should have another internal function that performs the same checks as _checkProtocolFee since its copied just twice here. Want to reduce repeated code paths bc it opens surface area for bugs

Reply:
yea agree, got rid of checkProtocolFee initially because it was the old behavior (reverting) but can re-add to reduce duplication